File: KnapsackProblem/Python/KnapsackProblem.py
def knapsackProblem(someListOfLists, index, capacity):
    
    weight = someListOfLists[index][0]
    price = someListOfLists[index][1] 
    if index == 0 or capacity == 0:
        return 0
    elif weight > capacity:
        return knapsackProblem(someListOfLists, index - 1, capacity)
    else:
        tempOne = knapsackProblem(someListOfLists, index - 1, capacity)
        tempTwo = price + knapsackProblem(someListOfLists, index - 1, capacity - weight)
        result = max(tempOne, tempTwo)
        return result
                                    

listOfListsOne = [[1, 5], [2, 3], [4, 3], [2, 3], [5, 2]]
listOfListsTwo = [[10, 60], [20, 100], [30, 120]]
print(knapsackProblem(listOfListsOne, len(listOfListsOne) - 1, 10))
print(knapsackProblem(listOfListsTwo, len(listOfListsTwo) - 1, 50))

File: KnapsackProblem/C#/KnapsackProblem.cs
using System;

namespace KnapsackProblem
{
	class Program
	{
		public static int Max(int a, int b)
		{
			return (a > b) ? a : b;
		}

		public static int KnapsackProblem(int [,] someArr, int index, int capacity)
		{
			if(index == 0 || capacity == 0)
				return 0;
			else if(someArr[index, 0] > capacity)
				return KnapsackProblem(someArr, index - 1, capacity);
			else
			{
				int tempOne = KnapsackProblem(someArr, index - 1, capacity);
				int tempTwo = someArr[index, 1] + KnapsackProblem(someArr, index - 1, capacity - someArr[index, 0]);
				return Max(tempOne, tempTwo);
		
			}
		}

		public static void Main(string[] args)
		{
			int [,] weightsAndValuesOne = new int [,] {{1, 5}, {2, 3}, {4, 5}, {2, 3}, {5, 2}};
			int [,] weightsAndValuesTwo = {{10, 60}, {20, 100}, {30, 120}};

			Console.WriteLine(KnapsackProblem(weightsAndValuesOne, weightsAndValuesOne.GetLength(0) - 1, 10));
			Console.WriteLine(KnapsackProblem(weightsAndValuesTwo, weightsAndValuesTwo.GetLength(0) - 1, 50));
		}
	}
}

